### ISSUE TITLE
Integrate DYDX into: data_factory, trader, sim, and predictoor

### ISSUE THREAD

**Issue by idiom-bytes** [oceanprotocol/pdr-backend#492]
### Motivation
A new venue to trade that is available to US customers. A first step was simply to buy the aggregate BTCUSD subscription to trade on dydx, a next step may be to do a full integration such that it can handle a locally-trained predictive-trading model.

### Requirements
An integration that covers: price feeds, sim engine, predictions, and trading. All new code for dydx integrations requires sample data, unit test, coverage, etc...

#### pdr_backend/lake/
Please look at safe_fetch_ohlcv(), @trentmc pointed towards this. This is a good place to call DYDX, fetch the data, and retrieve it for the data factory.

This function is called by ohlcv_data_factory, and you can see that it even supports a limit=1000 parameter, such that you can make sure that DYDX is handled correctly fetch_limit = 100 if exch == 'DYDX' else 1000... perhaps moving this into the safe_fetch_ohlcv() so the function sorts out its own dependencies.

On that note, there might be other dependencies to update to integrate DYDX, such as exch related checks and hooks to ohlcv_data_factory.py such that we can properly use the yaml config, and such that everything relating to dydx "just works (tm)".

#### pdr_backend/sim/
You'll have to make sure that the data feed integration is working, and that merge_df is returning a dataframe that contains dydx data, and that the sim is working correctly against it

#### pdr_backend/predictoor/
You have already verified the sim works. Predictoor should be able to, as well.

#### pdr_backend/trader/
You'll have to update the pdr trader agent such that it's able to connect to DYDX exchange, and issue buy/sell orders through there.

### DoD
- [ ] ohlcv integration
- [ ] sim integration
- [ ] predictoor integration
- [ ] trader integration
- [ ] dydx is working end-to-end across all modules, such that the user is able to fetch data into the lake, simulate predictions with dydx, and then trade based on a locally computed prediction

**Comment by trentmc** [comment]
I just added the following into the top of the description:

----

### Pre-requisite

[pdr-private#65](https://github.com/oceanprotocol/pdr-private/issues/65] "[Sim, Make $] Explore dYdX" came before this issue.

Its main goal is to find out: "can we make $ with dydx"? Only once we have completed that and got a "yes", should we integrate dydx everywhere (predictoor, trader, etc)

----

I have also re-opened [pdr-private#65](https://github.com/oceanprotocol/pdr-private/issues/65].

**Comment by idiom-bytes** [comment]
@trentmc I have reviewed the description/outline of this ticket as discussed. I also avoided creating additional tickets so we can take this step-by-step. I went ahead and closed pdr-private.

Thank you

**Comment by idiom-bytes** [comment]
Hi @graceful-coder, I noticed the following were marked as completed from issue #65.
```
- [x] make a new branch from yaml-cli2 branch. Work in this new branch for remaining work here
- [x] add one dydx pair to simulator flow (pdr_backend/sim)
- [x] run simulation on the pair.
Be sure to have the correct % tx fee. (It's an input to simulator flow, via the ppss.yaml file.)
What's $ made vs time? Accuracy vs time? (Those plots are output from sim)
```

Please point me to the branch when you can so I can take a look. I would consider renaming that branch to `issue492-epic-exploring-dydx` (this ticket) so you can continue merging PRs to it until you have a solid DYDX implementation.

**Comment by trentmc** [comment]
> I don't think this is worth pursuing:
> 
> * Binance futures has lower fees than dydx
> * More liquidity in Binance
> * Faster trading on Binance
> 
> @trentmc @idiom-bytes

I'm OK open to closing this.

But in lieu of it, how about we create a new issue for: can run trader bot on binance futures

**Comment by trentmc** [comment]
Let's discuss in that issue (910)
